theme/desktop/kde.go

```diff
@@ -0,0 +1,206 @@
+func (k *KDETheme) Icon(i fyne.ThemeIconName) fyne.Resource {
+	return ft.DefaultTheme().Icon(i)
+}
+
+// Size returns the size of the font for the specified text style.
+//
+// Implements: fyne.Theme
+func (k *KDETheme) Size(s fyne.ThemeSizeName) float32 {
+	if s == ft.SizeNameText {
+		return k.fontSize
+	}
+	return ft.DefaultTheme().Size(s)
+}
+
+// decodeTheme initialize the theme.
+func (k *KDETheme) decodeTheme() error {
+	if err := k.loadScheme(); err != nil {
+		return err
+	}
+	k.setFont()
+	return nil
+}
+
+// loadScheme loads the KDE theme from kdeglobals if it is found.
+func (k *KDETheme) loadScheme() error {
+	// the theme name is declared in ~/.config/kdedefaults/kdeglobals
+	// in the ini section [General] as "ColorScheme" entry
+	homedir, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+	content, err := ioutil.ReadFile(filepath.Join(homedir, ".config/kdeglobals"))
+	if err != nil {
+		return err
+	}
+
+	section := ""
+	lines := strings.Split(string(content), "\n")
+	for _, line := range lines {
+		if strings.HasPrefix(line, "[") {
+			section = strings.ReplaceAll(line, "[", "")
+			section = strings.ReplaceAll(section, "]", "")
+		}
+		if section == "Colors:Window" {
+			if strings.HasPrefix(line, "BackgroundNormal=") {
+				k.bgColor = k.parseColor(strings.ReplaceAll(line, "BackgroundNormal=", ""))
+			}
+			if strings.HasPrefix(line, "ForegroundNormal=") {
+				k.fgColor = k.parseColor(strings.ReplaceAll(line, "ForegroundNormal=", ""))
+			}
+		}
+		if section == "Colors:Button" {
+			if strings.HasPrefix(line, "BackgroundNormal=") {
+				k.buttonColor = k.parseColor(strings.ReplaceAll(line, "BackgroundNormal=", ""))
+			}
+			if strings.HasPrefix(line, "BackgroundAlternate=") {
+				k.buttonAlternate = k.parseColor(strings.ReplaceAll(line, "BackgroundAlternate=", ""))
+			}
+		}
+		if section == "Colors:View" {
+			if strings.HasPrefix(line, "BackgroundNormal=") {
+				k.viewColor = k.parseColor(strings.ReplaceAll(line, "BackgroundNormal=", ""))
+			}
+		}
+		if section == "General" {
+			if strings.HasPrefix(line, "font=") {
+				k.fontConfig = strings.ReplaceAll(line, "font=", "")
+			}
+		}
+	}
+
+	return nil
+}
+
+// parseColor parses a color from a string in form r,g,b or r,g,b,a.
+func (k *KDETheme) parseColor(col string) color.Color {
+	// the color is in the form r,g,b,
+	// we need to convert it to a color.Color
+
+	// split the string
+	cols := strings.Split(col, ",")
+	// convert the string to int
+	r, _ := strconv.Atoi(cols[0])
+	g, _ := strconv.Atoi(cols[1])
+	b, _ := strconv.Atoi(cols[2])
+	a := 0xff
+	if len(cols) > 3 {
+		a, _ = strconv.Atoi(cols[3])
+	}
+
+	// convert the int to a color.Color
+	return color.RGBA{uint8(r), uint8(g), uint8(b), uint8(a)}
+}
+
+// setFont sets the font for the theme.
+func (k *KDETheme) setFont() {
+
+	if k.fontConfig == "" {
+		return
+	}
+	// the fontline is in the form "fontline,size,...", so we can split it
+	fontline := strings.Split(k.fontConfig, ",")
```

Comment:
I think you might want to use `strings.SplitN` here given that you only care about the two first ones